src/index.ts

```diff
@@ -0,0 +1,59 @@
+import WebSocket from 'ws';
+import { BaseMill } from '@open-oracle-origami/origami-js-sdk';
+
+class CoinbaseMill extends BaseMill {
+  private ws: WebSocket;
+  private readonly pairs: string[];
+
+  constructor({
+    id = 'coinbase',
+    socketServerUrl = 'wss://ws-feed.exchange.coinbase.com',
+    // @ts-ignore
+    pairs = ['BTC-USDT'],
+  }) {
+    super();
+
+    if (id) this.setId(id);
+    this.pairs = pairs;
```

Comment:
Debating if we call these collections, I guess technically it's not origami however.

Reply:
yeah this is internal only to the mill and should be accessible outside like this